synapse/util/caches/lrucache.py

```diff
@@ -271,7 +271,10 @@ def drop_from_cache(self) -> None:
         removed from all lists.
         """
         cache = self._cache()
-        if cache is None or cache.pop(self.key, None) is None:
+        if (
+            cache is None
+            or cache.pop(self.key, _Sentinel.sentinel) is _Sentinel.sentinel
```

Comment:
So that we can store `None` as a value in the cache?

Reply:
That was the thinking, yes.

Although `drop_from_lists` is idempotent so it's functionally the same